contentcuration/contentcuration/static/js/edit_channel/new_channel/views.js

```diff
@@ -304,6 +341,91 @@ var ChannelListItem = BaseViews.BaseListEditableItemView.extend({
 	},
 });
 
+var PendingChannelList  = BaseViews.BaseEditableListView.extend({
+	template: require("./hbtemplates/channel_list_pending.handlebars"),
+	list_selector: "#channel_list_pending",
+	default_item: ".default-item",
+
+	initialize: function(options) {
+		this.bind_edit_functions();
+		this.container = options.container;
+		this.collection = new Models.InvitationCollection();
+		this.render();
+	},
+	render: function() {
+		this.$el.html(this.template());
+		var self = this;
+		window.current_user.get_pending_invites().then(function(invitations){
+			self.collection.reset(invitations.toJSON());
+			self.load_content(self.collection, " ");
+		});
+	},
+	create_new_view:function(data){
+		var newView = new ChannelListPendingItem({
+			model: data,
+			containing_list_view: this,
+		});
+		this.views.push(newView);
+  		return newView;
+	},
+	invitation_submitted: function(invitation, channel){
+		this.collection.remove(invitation);
+		if(channel){
+			this.container.add_channel(channel);
+		}
+	}
+});
+
+var ChannelListPendingItem = BaseViews.BaseListEditableItemView.extend({
+	tagName: "li",
+	id: function(){
+		return (this.model)? this.model.get("id") : "new";
+	},
+	className:"pending_container",
+	template: require("./hbtemplates/channel_item_pending.handlebars"),
+	initialize: function(options) {
+		this.bind_edit_functions();
+		_.bindAll(this, 'accept','decline', 'submit_invitation');
+		this.listenTo(this.model, "sync", this.render);
+		this.containing_list_view = options.containing_list_view;
+		this.status = null;
+		this.render();
+	},
+	render: function() {
+		this.$el.html(this.template({
+			invitation: this.model.toJSON(),
+			status: this.status
+		}));
+	},
+	events: {
+		'click .accept_invite':'accept',
+		'click .decline_invite':'decline'
+	},
+	accept: function(){
+		var self = this;
+		this.model.accept_invitation().then(function(channel){
+			self.submit_invitation(true, channel);
+		}).catch(function(error){
+			dialog.alert("Invitation Error", error.responseText, null);
+        });
+	},
+	decline: function(){
+		var self = this;
+		dialog.dialog("Declining Invitation", "Are you sure you want to decline this invitation?", {
+            "CANCEL":function(){},
```

Comment:
Are these options for [JQuery's Dialog](http://api.jqueryui.com/dialog/)? Wouldn't it be better to pass in button options this arrangement, with empty functions?